fhir-notification-nats/src/main/java/com/ibm/fhir/notifications/nats/impl/FHIRNotificationNATSPublisher.java

```diff
@@ -0,0 +1,157 @@
+import io.nats.streaming.StreamingConnection;
+import io.nats.streaming.StreamingConnectionFactory;
+
+import com.ibm.fhir.notification.FHIRNotificationEvent;
+import com.ibm.fhir.notification.FHIRNotificationService;
+import com.ibm.fhir.notification.FHIRNotificationSubscriber;
+import com.ibm.fhir.notification.exception.FHIRNotificationException;
+import com.ibm.fhir.notification.util.FHIRNotificationUtil;
+
+/**
+ * This class implements the FHIR server notification service via a Kafka topic.
+ */
+public class FHIRNotificationNATSPublisher implements FHIRNotificationSubscriber {
+    private static final Logger log = Logger.getLogger(FHIRNotificationNATSPublisher.class.getName());
+    private static FHIRNotificationService service = FHIRNotificationService.getInstance();
+
+    private StreamingConnection sc = null;
+    private AckHandler acb = null;
+    private String channelName = null;
+
+    // "Hide" the default ctor.
+    protected FHIRNotificationNATSPublisher() {
+    }
+
+    public FHIRNotificationNATSPublisher(String clusterId, String channelName, String server) {
+        log.entering(this.getClass().getName(), "ctor");
+        try {
+            init(clusterId, channelName, server);
+        } finally {
+            log.exiting(this.getClass().getName(), "ctor");
+        }
+    }
+
+    /**
+     * Performs any required initialization to allow us to publish events to the topic.
+     */
+    private void init(String clusterId, String channelName, String server) {
+        log.entering(this.getClass().getName(), "init");
+
+        try {
+            this.channelName = channelName;
+            String clientId = "fhir-server";
+            if (log.isLoggable(Level.FINER)) {
+                log.finer("ClusterId: " + clusterId);
+                log.finer("Channel name: " + channelName);
+                log.finer("Server: " + server);
+            }
+
+            // Make sure that the properties file contains the servers property at a minimum.
+            if (server == null || clusterId == null) {
+                throw new IllegalStateException("Config property missing from the NATS connection properties.");
+            }
+
+            // Create the NATS connection
+            Options streamingOptions = new Options.Builder().natsUrl(server).clusterId(clusterId).clientId(clientId).build();
+            StreamingConnectionFactory cf = new StreamingConnectionFactory(streamingOptions);
+            sc = cf.createConnection();
+
+            // Create the publish callback
+            acb = new AckHandler() {
+                @Override
+                public void onAck(String nuid, Exception ex) {
+                    log.finer("Received ACK for guid: " + nuid);
+                    if (ex != null) {
+                        log.log(Level.SEVERE, "Error in server ack for guid " + nuid + ": " + ex.getMessage(), ex);
+                    }
+                    // TODO: Do something else here
```

Comment:
What's the TODO for... what's the something else?

Reply:
We need to answer the question of what we want to do on failure.  Resubmit on failure or just log?  That TODO is probably misplaced, as we may not want to do anything further on success.

Reply:
Opted to not do anything additional in this implementation.  As we work more with NATS, what to do on failure, if anything, should become clearer.  Removed TODO.